### PR TITLE
Add delete support and handwriten sweeper for Firebase Web App

### PR DESCRIPTION
**Pull request by rainshen49**
Support DELETE for Firebase Web apps. By default the delete is soft with a 30 days expiration. For the sweeper, there is no point keeping test apps for 30 days thus the hard-delete.

If this PR is for Terraform, I acknowledge that I have:

- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
firebase: added deletion support for `google_firebase_web_app`
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
I notice this pr is in draft. Let me know when it's ready for review

File: mmv1/products/firebase/terraform.yaml [hunk 1]
```diff
@@ -41,8 +41,7 @@ overrides: !ruby/object:Overrides::ResourceOverrides
   WebApp: !ruby/object:Overrides::Terraform::ResourceOverride
     import_format: ['{{project}} {{name}}']
     autogen_async: true
-    skip_delete: true #currently only able to delete a webapp through the Firebase Admin console
-    skip_sweeper: true
+    skip_sweeper: true # handwritten sweeper to hard-delete
```
Comment (tylerg-dev) [inline mmv1/products/firebase/terraform.yaml:44]: is the hand-written sweeper not in this PR?

Comment (rainshen49) [reply mmv1/products/firebase/terraform.yaml:44]: Yes its the file below. `skip_sweeper` means "don't generate a sweeper". Kind of a bad name there.

**Comment by ScottSuarez** [comment]
Hi Rainshen, I notice this PR is in draft. Let me know when it's ready for review.

**Comment by rainshen49** [comment]
Question: is the sweeper run as part of the tests? From the debug log, I don't see a request with url `:remove?immediate=true`

**Comment by rainshen49** [comment]
Hi @ScottSuarez this PR is ready for review

File: mmv1/products/firebase/api.yaml [hunk 2]
```diff
@@ -132,6 +134,16 @@ objects:
...
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'deletion_policy'
+        description: |
+          Optional. Set to `ABANDON` to soft-delete the WebApp. Soft-deleted WebApps are recoverable
+          within 30 days using [projects.webApps.undelete](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects.webApps/undelete).
+          Set to `DELETE` to immediately delete the WebApp. Default to `DELETE`
+        input: true
+        url_param_only: true
+        default_value: DELETE
```
Comment (ScottSuarez) [inline mmv1/products/firebase/api.yaml:146]: This is a pretty significant change. The resource behavior is going from un-delete-able to default delete-able. Should the default be abandon with delete being set within our tests?

Comment (ScottSuarez) [reply mmv1/products/firebase/api.yaml:146]: What are the other valid options for this?

File: mmv1/templates/terraform/examples/firebase_web_app_basic.tf.erb [hunk 3]
```diff
@@ -15,6 +15,7 @@ resource "google_firebase_web_app" "<%= ctx[:primary_resource_id] %>" {
 	provider = google-beta
 	project = google_project.default.project_id
 	display_name = "<%= ctx[:vars]['display_name'] %>"
+    deletion_policy = "DELETE"
```
[the remark this answers is not in the document]
Comment (rainshen49) [reply mmv1/templates/terraform/examples/firebase_web_app_basic.tf.erb]: 😢 Done

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**